### ISSUE TITLE
search, filter

### ISSUE THREAD

**Issue by jennybc** [timelyportfolio/listviewer#2]
@timelyportfolio raised this question on twitter … yes it would be nice to be able to search or filter the list being inspected. This could be taken literally or maybe just control the initial state of the expansion toggle I see in the example based on `par()`.

This new CRAN package [`searchable`](http://cran.r-project.org/web/packages/searchable/index.html) has intriguing functionality.

It feels like an XPath-like syntax for communicating which parts of the list are interesting would be worth exploring.

**Comment by timelyportfolio** [comment]
Very interesting as a general topic.  I was not aware of `searchable`, so will explore it.  Also, I think you are aware of the powerful searching/filtering, etc. from @renkun-ken `rlist` http://renkun.me/rlist-tutorial/Features/Filtering.html.

**Comment by jennybc** [comment]
I should try the rlist filtering functions. They seem more focused on filtering by *content* than by name or attributes. I need to specify name filters, several layers deep, by regex, hence the idea for XPath-ish syntax. But maybe rlist does this...

**Comment by timelyportfolio** [comment]
You making good progress?  Happy to try to answer questions.  `rlist` is full of all sorts of potential and just needs a healthy dose of imagination.

**Comment by timelyportfolio** [comment]
should reference https://github.com/renkun-ken/rlist/issues/104#issuecomment-146799151 and https://github.com/renkun-ken/rlist/issues/110

**Comment by randomgambit** [comment]
@jennybc @timelyportfolio I am not quite sure this is what Jenny proposed, but it would be very nice to be able to highlight the matched terms when one uses the search function.

For instance, in each row of my dataframe I store several paragraphs of texts. Now using the search function for a specific term in listviewer will highlight the whole paragraph, but not the exact word in the paragraph. That is annoying when one checks if regex functions were counting the words properly, etc

What do you think?
Thanks!

**Comment by randomgambit** [comment]
@jennybc @timelyportfolio does that make sense? thanks!!

**Comment by timelyportfolio** [comment]
@randomgambit yes definitely makes sense and is a great idea.  I will not have time to work on it unfortunately for a couple of weeks though.  I am guessing this could be accomplished with https://markjs.io/ but will need to explore more.

**Comment by randomgambit** [comment]
@timelyportfolio that would be amazing. I think this package is absolutely amazing. I use pretty large datasets that contain text, timestamps and `jsonedit` is the only way to explore the data without messing with the print format in the console.

**Comment by timelyportfolio** [comment]
I thought this would be fairly straightforward, but unfortunately without a good event stream from jsoneditor, the asychrony and dom manipulation results in a very poor outcome.  I will need some more time and thought to successfullly tackle this one :(

**Comment by randomgambit** [comment]
hello, any update on this? are you aware of other similar data content visualization tools?
Thanks!

**Comment by timelyportfolio** [comment]
@randomgambit, I never figured this one out.  Hoping the new React version of jsoneditor will help address this, but this version seems like it will not be in the near future.  I'll stay on the lookout for other libraries to solve this.  As of now I do not know of anything.  Sorry.

**Comment by randomgambit** [comment]
ok.. thanks for helping out anyway. in any case, there is no other list viewer like your at the moment, right?

**Comment by randomgambit** [comment]
Hi @jennybc @timelyportfolio how are you? just checking.. any update on this? This package is amazing and I truly believe this feature is the only missing piece... highlighting the search terms into each cell would be amazing!! I saw a new update of the package. are there any good news here? :) thanks!!
